lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
I think it would be less magical to just use `Reflection#foreign_key` which will return the option or it determines the name. That sounds more robust than just checking the options as it will always try to return something even if it is name generated by rails.

Looks like there is a method from Rails `reflect_on_all_associations` that accepts macro. So it can be further simplified:

```
reflections = @object.class.try(: reflect_on_all_associations, :belongs_to) || []
reflections.map { |r| r.foreign_key.to_sym }
```